packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -32,7 +31,13 @@ import {
 	prepareExecutionDataForDbUpdate,
 	updateExistingExecution,
 } from './shared/shared-hook-functions';
-import { toSaveSettings } from './to-save-settings';
+import { type ExecutionSaveSettings, toSaveSettings } from './to-save-settings';
+
+type HooksSetupParameters = {
+	saveSettings: ExecutionSaveSettings;
+	pushRef?: string;
+	retryOf?: string;
+};
```

Comment:
Does it make sense to remove this type and use the prop types directly?

Usage hints that these props might not belong together:
- `hookFunctionsPush` accepts `saveSettings` and does not use it
- `hookFunctionsSaveProgress` accepts`pushRef` and `retryOf` and does not use them
- `hookFunctionsSave` uses all three props
- `hookFunctionsSaveWorker` accepts `saveSettings` and does not use it
- `getWorkflowHooksWorkerExecuter` inline omits `saveSettings` so we only pass it the other two
- `getWorkflowHooksWorkerMain` inline omits `saveSettings` so we only pass it the other two

Reply:
All this code will be re-written once I convert this file to switch to DI, like I did in #12364